x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -212,7 +213,16 @@ func (s *scheduler) addFailedJobs(ctx context.Context, jobs []*job) []*job {
 		if !jobMap[name] {
 			obj, err := s.bucket.Object(name).Attrs(ctx)
 			if err != nil {
-				s.log.Errorf("adding failed job %s to job list caused an error: %v", name, err)
+				if errors.Is(err, storage.ErrObjectNotExist) {
+					// if the object is not found in the bucket, then remove it from the failed job list
+					s.state.deleteFailedJob(name)
+					s.log.Debugf("scheduler: failed job %s not found in bucket %s", name, s.src.BucketName)
```

Comment:
Is debug an appropriate level? How frequently do we expect this to happen?

Reply:
This depends on the users TTL settings, in some scenarios if the user is unaware of their TTL, this could be pretty often.